### DISCUSSION TITLE
Guide for Production / Hardening

### DISCUSSION THREAD

**Discussion by lecramr** [coollabsio/coolify#2674] [General]
Hey,
Are there any resources on how to secure a coolify instance? As I am currently hosting it on a Root-Server without a HW-Firewall in front. As it seems Coolify is somehow messing with ufw-docker, and I can not get it to work when ufw is enabled. Anybody has already a solution? It doesn't feel right to have the instance exposed without even ufw.

Maybe even a feature idea, so setup ufw together with coolify out of the box.

Thanks in advance!

**Comment by engineervix** [comment]
I came across https://blog.arcjet.com/protecting-self-hosted-coolify-apps/ which discusses using tailscale.
